rust/lib/src/lib.rs

```diff
@@ -185,8 +193,29 @@ pub fn save_to_b64() -> String {
     }
 }
 
+fn set_server(server: String) -> String {
+    if let Some(server) = server.try_into().ok() {
+        if let Ok(mut lc) = LIGHTCLIENT.lock() {
+            if let Some(lc) = &mut lc.get_mut() {
+                Arc::get_mut(lc).unwrap().set_server(server);
+                "server set"
+            } else {
+                "can't get arc"
+            }
+        } else {
+            "can't get mutex"
+        }
+    } else {
+        "bad server uri"
+    }
+    .to_string()
+}
+
 pub fn execute(cmd: String, args_list: String) -> String {
     let resp: String;
+    if cmd == String::from("changeserver") {
+        return set_server(args_list);
+    }
```

Comment:
I'm wondering if between those } and { is missing an else???? This structure is weird to me, but I don't know anything about rust.

Reply:
The else isn't needed because if we hit the `return` then the rest of the function won't be run, but an else would probably make it clearer.